torchvision/models/regnet.py

```diff
@@ -0,0 +1,672 @@
+        activation_layer: Callable[..., nn.Module],
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(OrderedDict(
+            a=ConvBNActivation(width_in, width_out, kernel_size=3, stride=stride,
+                               norm_layer=norm_layer, activation_layer=activation_layer),
+            b=ConvBNActivation(width_out, width_out, kernel_size=3, stride=1,
+                               norm_layer=norm_layer, activation_layer=activation_layer),
+        ))
+
+
+class ResBasicBlock(nn.Module):
+    """Residual basic block: x + F(x), F = basic transform."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__()
+        self.proj_block = (width_in != width_out) or (stride != 1)
+        if self.proj_block:
+            self.proj = nn.Conv2d(
+                width_in, width_out, 1, stride=stride, padding=0, bias=False
+            )
+            self.bn = norm_layer(width_out)
+        self.f = BasicTransform(
+            width_in, width_out, stride, norm_layer, activation_layer
+        )
+        self.activation = activation_layer(inplace=True)
+
+    def forward(self, x: Tensor) -> Tensor:
+        if self.proj_block:
+            x = self.bn(self.proj(x)) + self.f(x)
+        else:
+            x = x + self.f(x)
+
+        return self.activation(x)
+
+
+class BottleneckTransform(nn.Sequential):
+    """Bottleneck transformation: 1x1, 3x3 [+SE], 1x1."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+        group_width: int,
+        bottleneck_multiplier: float,
+        se_ratio: Optional[float],
+    ) -> None:
+        layers = OrderedDict()
+        w_b = int(round(width_out * bottleneck_multiplier))
+        g = w_b // group_width
+
+        layers["a"] = ConvBNActivation(width_in, w_b, kernel_size=1, stride=1,
+                                       norm_layer=norm_layer, activation_layer=activation_layer)
+        layers["b"] = ConvBNActivation(w_b, w_b, kernel_size=3, stride=stride, groups=g,
+                                       norm_layer=norm_layer, activation_layer=activation_layer)
+
+        if se_ratio:
+            # The SE reduction ratio is defined with respect to the
+            # beginning of the block
+            width_se_out = int(round(se_ratio * width_in))
+            layers["se"] = SqueezeExcitation(
+                input_channels=w_b,
+                squeeze_channels=width_se_out,
+                activation=activation_layer,
+            )
+
+        layers["c"] = nn.Conv2d(w_b, width_out, 1, stride=1, padding=0, bias=False)
+        layers["final_bn"] = norm_layer(width_out)
```

Comment:
@mannatsingh / @blefaudeux does the layer name(a, b, c final_bn) have special meanings? otherwise we could  fuse "c" and "final_bn" into one `ConvBN` op.

Reply:
Nit: This could be replaced by ConvBNActivation. See my other comment on why `final_bn` can be useful to know.